staging/src/k8s.io/kube-aggregator/pkg/apiserver/openapi_aggregator.go

```diff
@@ -95,40 +96,62 @@ func newOpenAPIAggregator(delegateHandler http.Handler, webServices []*restful.W
 // openAPISpecInfo is used to store OpenAPI spec with its priority.
 // It can be used to sort specs with their priorities.
 type openAPISpecInfo struct {
-	serviceName string
-	spec        *spec.Swagger
-	priority    int32
+	apiService apiregistration.APIService
+	spec       *spec.Swagger
 }
 
 // byPriority can be used in sort.Sort to sort specs with their priorities.
-type byPriority []openAPISpecInfo
+type byPriority struct {
+	specs           []openAPISpecInfo
+	groupPriorities map[string]int32
+}
 
-func (a byPriority) Len() int      { return len(a) }
-func (a byPriority) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a byPriority) Len() int      { return len(a.specs) }
+func (a byPriority) Swap(i, j int) { a.specs[i], a.specs[j] = a.specs[j], a.specs[i] }
 func (a byPriority) Less(i, j int) bool {
-	if a[i].priority != a[j].priority {
-		// Sort by priority, higher first
-		return a[i].priority > a[j].priority
+	var iPriority, jPriority int32
+	if a.specs[i].apiService.Spec.Group == a.specs[j].apiService.Spec.Group {
+		iPriority = a.specs[i].apiService.Spec.VersionPriority
+		jPriority = a.specs[i].apiService.Spec.VersionPriority
 	} else {
-		// Sort by service name.
-		return a[i].serviceName < a[j].serviceName
+		iPriority = a.groupPriorities[a.specs[i].apiService.Spec.Group]
```

Comment:
I fear our logic there is more complicated. Look for some quiet moment and read what we came up with shortly before 1.7 release: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1beta1/types.go#L59 ;-) Wondering, can we expose that logic from kube-aggregator code somehow and reuse it here?